### ISSUE TITLE
Train on single GPU

### ISSUE THREAD

**Issue by liuxy416** [CoinCheung/BiSeNet#2]
Thanks for your sharing. Did you try to train BiSeNet using one GPU? Besides, could you please share your model on cityscapes dataset?

**Comment by CoinCheung** [comment]
No, I did not use one gpu to train, since this model involves BN which requires the batch size to be big enough, and using one gpu might not satisfy this batch size requirement. I will be happy to release my trained model, but not these days. Maybe I will do it in next weeks.

**Comment by CoinCheung** [comment]
You may have a try of training with one gpu. Just double the training iters and half the learning rate, and see if the result is the same.

**Comment by CoinCheung** [comment]
Hi, 

I do not know if you have tried to train on single gpu, but I have uploaded the pretrained models. If you still need them, please follow the links in the `README.md` file.

**Comment by liuxy416** [comment]
@CoinCheung Sorry for the late replay, I have trained this model on one GPU a few days ago with batch size=8 and iterations=80K, and i finally got mIOU=72.9.  I think the main gap comes from the number of GPUs, I will try on two GPUs this weekend, then I'll get back to you. 

Besides,  I tried to modify the architecture of BiseNet recently, I simply added two ASPP module (which comes from Deeplab v3) before ARM to enlarge the receptive field and get more context-level information, but it didn't work very well. Do you have any suggestions?

Looking forward to your reply！

**Comment by CoinCheung** [comment]
Hi, 

Did you half the learning rate when training with one gpu?

As for the aspp problem, my comprehension is that: aspp and the  u-shape decoder-encoder structure contribute to the model performance in the way. They are both designed to cope with the mult-scale problem. Aspp extracts features of different scaled receipt fields with multi parallel atrous convolutions, while the u-shape structure directly uses features from different layers in the backbone. The different layers that u-shape structure uses usually have different spatial shape, which introduces information with different receipt field. Since aspp and u-shape structure have overlapping contributions to the model performance, you are thus not expected to observe a further boost with aspp added to the context path of bisenet.  

That is my personal understanding, I am not sure if it is correct. Hope this would help.

**Comment by liuxy416** [comment]
> Hi,
> 
> Did you half the learning rate when training with one gpu?
> 
> As for the aspp problem, my comprehension is that: aspp and the u-shape (decoder-encoder structure) contribute to the model performance in similar way. They are both designed to cope with the mult-scale problem. Aspp extracts features of different scaled receipt fields with multi parallel atrous convolutions, while the u-shape structure directly uses features from different layers in the backbone. The different layers that u-shape structure uses usually have different spatial shape, which introduces information with different receipt field. Since aspp and u-shape structure have overlapping contributions to the model performance, you are thus not expected to observe a further boost with aspp added to the context path of bisenet.
> 
> That is my personal understanding, I am not sure if it is correct. Hope this would help.

Thank you for your help, I'll try training with half lr this week. 
I agree with you on the ASPP part, and I think I need to change my strategies on architecture modification.

**Comment by REFunction** [comment]
I have trained with half learning rate and double iters on one gpu.
I get 78.5 mIoU

**Comment by CoinCheung** [comment]
I am closing this, since  one gpu training is verified.

**Comment by codermckee** [comment]
> I have trained with half learning rate and double iters on one gpu.
> I get 78.5 mIoU
 Hi, How long did it take you to get final result?

**Comment by codermckee** [comment]
> I have trained with half learning rate and double iters on one gpu.
> I get 78.5 mIoU

@REFunction  Hi, How long did it take you to get final result?

**Comment by REFunction** [comment]
> > I have trained with half learning rate and double iters on one gpu.
> > I get 78.5 mIoU
> 
> @REFunction Hi, How long did it take you to get final result?

About 1.5 days
